File: backend/app/schemas/insights.py
from datetime import datetime
from pydantic import BaseModel


class InsightItem(BaseModel):
    id: str
    text: str
    type: str  # warning / info / success


class InsightsResponse(BaseModel):
    insights: list[InsightItem]
    generated_at: datetime | None = None
    cached: bool = False

File: frontend/src/api/insights.ts
import { api } from './client'

export interface Insight {
  id: string
  text: string
  type: 'warning' | 'info' | 'success'
}

export interface InsightsResponse {
  insights: Insight[]
  generated_at: string | null
  cached: boolean
}

export const getInsights = (force = false) =>
  api.get<InsightsResponse>('/api/insights', { params: force ? { force: true } : {} })
